jsonrpc/codec.go

```diff
@@ -127,13 +138,48 @@ const (
 	EarliestBlockNumber = BlockNumber(-1)
 )
 
+var (
+	PendingBlockNumberOrHash  BlockNumberOrHash
+	LatestBlockNumberOrHash   BlockNumberOrHash
+	EarliestBlockNumberOrHash BlockNumberOrHash
+)
+
 type BlockNumber int64
 
+// String returns the string representation of the block number
+func (b *BlockNumber) String() string {
+	blockNumber := *b
+
+	switch blockNumber {
+	case PendingBlockNumber:
+		return pending
+	case LatestBlockNumber:
+		return latest
+	case EarliestBlockNumber:
+		return earliest
+	}
+
+	if *b < 0 {
+		panic("internal. blocknumber is negative") //nolint:gocritic
+	}
```

Comment:
Let's not protect here (especially not with throwing panic), just convert it to the string whatever it is.

Reply:
Fix: https://github.com/Ethernal-Tech/blade/pull/160/commits/0fa0ad9d68ba8a9d053db853491e907701b971db